core/src/main/codegen/templates/Parser.jj

```diff
@@ -238,6 +238,39 @@ public class ${parser.class} extends SqlAbstractParserImpl
         return SqlStmtList();
     }
 
+    // Q: Is it the best way ???
```

Comment:
I'd add a method `parseArray` to `SqlAbstractParserImpl`, as a top-level entry point to the parser. 

Also add a `parseArrayLiteral` method in `SqlParserUtil` similar to `parseTimestampLiteral`. The implementation will create a fresh parser then call `parseArray`.

If the validator needs to convert a char literal into an array literal, it can call that method.

Reply:
I would not preemptively convert every char literal into an array literal. Which means that `class SqlArrayCharStringLiteral` is probably not needed.

Reply:
Then I will need to pass ParserFactory to SqlValidatorImpl. If it is OK that validator will know about Parser then let's do like this.

Reply:
To do it properly, I agree, we should pass parser factory and config in the validator config. But for now it's probably OK to use the default parser.

Reply:
Done.